File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> indices = new HashMap<>();
        int n = nums.length;
        for(int i = 0; i < n; i++) {
            int diff = target - nums[i];
            if(indices.containsKey(diff)) {
                return new int[]{i, indices.get(diff)};
            }
            indices.put(nums[i], i);
        }
        return null;
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        l = []
        for i in range(len(nums)):
            p = target-nums[i]
            if(p in nums):
                ind = nums.index(p)
                if(i!=ind):
                    l = [i,ind]
                    return l